weather_mv/loader_pipeline/sinks.py

```diff
@@ -284,31 +291,176 @@ def __normalize_grib_dataset(filename: str) -> xr.Dataset:
+    """
+    Reads dataset and tiff_config given by user. By default if time or step dimension is present in source data
+    then it will partition the dataset based on this dimension. Based on tiff_config it will consider the dimensions
+    given by user to partition the dataset.
+    """
+
+    default_tiff_dims = set([x for x in ['time','step'] if ds.dims.get(x)])
+    dims = [ dim  for dim in ds.dims if dim not in ['latitude','longitude']]
+    coords_set = set(ds.coords.keys())
+    _merged_dataset_list = []
+    if tiff_config:
+        default_tiff_dims = default_tiff_dims.union(tiff_config['dims'])
+        if default_tiff_dims or dims:
+            flat_dims = set(dims).difference(default_tiff_dims)
+            _dims = [range(len(ds[x])) for x in dims]
+            if not default_tiff_dims:
+                default_tiff_dims = dims
+            groups = create_partition_configs(_dims,dims,default_tiff_dims)
+            for group in groups:
+                ds_attrs = ds.attrs
+                _data_array_list,dv_units_dict = create_partition(ds,group,flat_dims,coords_set)
+                ds_attrs['forecast_hour'] = _data_array_list[0].attrs['forecast_hour']
+                ds_attrs['start_time'] = _data_array_list[0].attrs['start_time']
+                ds_attrs['end_time'] = _data_array_list[0].attrs['end_time']
+                ds_attrs.update(**dv_units_dict)
+
+                merged_dataset = xr.merge(_data_array_list)
+                merged_dataset.attrs.clear()
+                merged_dataset.attrs.update(ds_attrs)
+                _merged_dataset_list.append(merged_dataset)
+
+            return _merged_dataset_list
+        else:
+            forecast_hour = int(ds.step.values / np.timedelta64(1, 'h')) if 'step' in coords_set else None
+
+            # We are going to treat the time field as start_time and the
+            # valid_time field as the end_time for EE purposes. Also, get the
+            # times into UTC timestrings.
+            start_time = _to_utc_timestring(ds.time.values)
+
+            end_time = _to_utc_timestring(ds.valid_time.values) if 'valid_time' in coords_set else start_time
+
+            attrs_to_add = {}
+            attrs_to_add['forecast_hour'] = forecast_hour  # Stick forecast hour in the metadata as well, that's useful.
+            attrs_to_add['start_time'] = start_time
+            attrs_to_add['end_time'] = end_time
+            ds.attrs.update(attrs_to_add)
+    return ds
+
+
 def __open_dataset_file(filename: str,
                         uri_extension: str,
                         disable_grib_schema_normalization: bool,
-                        open_dataset_kwargs: t.Optional[t.Dict] = None) -> xr.Dataset:
+                        open_dataset_kwargs: t.Optional[t.Dict] = None,
+                        tiff_config: t.Optional[t.Dict] = None) -> t.Union[xr.Dataset, t.List[xr.Dataset]]:
```

Comment:
I would like to avoid having `open_dataset` return a list of `datasets`, if possible.